types/state/stateTxn.go

```diff
@@ -0,0 +1,427 @@
+package state
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"github.com/AccumulateNetwork/accumulate/internal/logging"
+	"github.com/AccumulateNetwork/accumulate/smt/common"
+	"github.com/AccumulateNetwork/accumulate/smt/managed"
+	"github.com/AccumulateNetwork/accumulate/smt/pmt"
+	"github.com/AccumulateNetwork/accumulate/smt/storage"
+	"github.com/AccumulateNetwork/accumulate/smt/storage/database"
+	"github.com/AccumulateNetwork/accumulate/types"
+	"github.com/tendermint/tendermint/libs/log"
+	"sort"
+	"sync"
+	"time"
+)
+
+type DBTransactional struct {
+	db           *database.Manager
+	mm           *managed.MerkleManager
+	debug        bool
+	bpt          *pmt.Manager //pbt is the global patricia trie for the application
+	blockIndex   int64        //Index of the current block
+	TimeBucket   float64
+	mutex        sync.Mutex
+	updates      map[types.Bytes32]*blockUpdates
+	writes       map[storage.Key][]byte
+	transactions transactionLists
+	sync         sync.WaitGroup
+	logger       log.Logger
+}
+
+//func (tx *DBTransactional) init(debug bool) (err error) {
+//
+//	tx.debug = debug
+//	tx.updates = make(map[types.Bytes32]*blockUpdates)
+//	tx.writes = map[storage.Key][]byte{}
+//	tx.transactions.reset()
+//
+//	tx.bpt = pmt.NewBPTManager(tx.db)
+//	managed.NewMerkleManager(tx.db, markPower)
+//
+//	ent, err := tx.GetPersistentEntry(blockIndexKey[:], false)
+//	if err == nil {
+//		tx.blockIndex, _ = common.BytesInt64(ent.Entry)
+//	} else if !errors.Is(err, storage.ErrNotFound) {
+//		return err
+//	}
+//
+//	return nil
+//}
+
+func (s *StateDB) Begin() *DBTransactional {
+	dbTx := &DBTransactional{
+		db:           s.db,
+		logger:       s.logger,
+		bpt:          s.bpt,
+		mm:           s.mm,
+		debug:        s.debug,
+		mutex:        s.mutex,
+		TimeBucket:   s.TimeBucket,
+		sync:         s.sync,
+		updates:      s.updates,
+		transactions: s.transactions,
+		writes:       s.writes,
+		blockIndex:   s.blockIndex,
+	}
+	return dbTx
+}
+
+func (tx *DBTransactional) AddStateEntry(chainId *types.Bytes32, txHash *types.Bytes32, object *Object) {
```

Comment:
For now, change the receiver of the existing methods, do not create a new method. If you change the existing method, I can see the diff. If you add a new method, I can't easily compare that to the old method.